lib/core/sdam/topology.js

```diff
@@ -434,7 +436,7 @@ class Topology extends EventEmitter {
       }, serverSelectionTimeoutMS);
     }
 
-    this[kWaitQueue].push(waitQueueMember);
+    this[kWaitQueue].unshift(waitQueueMember);
```

Comment:
This changes the effective data structure to a stack since removal appears to always be made with shift. Is this intended? Should we make the naming more clear?

Reply:
it's a deque, so it can be a stack or a queue. In the `processWaitQueue` method below the data structure is indeed processed as a queue, but my expectation was that a server selection would want to occur immediately so I opted to unshift it here instead. This does make it a bit unfair in the face of _many_ incoming concurrent requests though, so I think I'll revert the change.

Reply:
reverted